src/core/mac/mac.cpp

```diff
@@ -257,6 +257,11 @@ bool Mac::IsEnergyScanInProgress(void)
     return (mState == kStateEnergyScan) || (mPendingScanRequest == kScanTypeEnergy);
 }
 
+bool Mac::IsInTransmitState(void)
+{
+    return (mState == kStateTransmitData) || (mState == kStateTransmitBeacon);
```

Comment:
This does not capture IEEE 802.15.4 Beacon Request transmissions during an active scan.  Granted, active scans are relatively rare and typically user initiated.  Thoughts?